functions/average.go

```diff
@@ -0,0 +1,12 @@
+package functions
+
+func (s SliceType) Average() float64 {
+	var sum ElementType
+	if len(s) == 0 {
+		return float64(0)
+	}
+	for _, i := range s {
+		sum += i
+	}
+	return float64(sum / ElementType(len(s)))
```

Comment:
This will not divide correctly, you are doing integer division if ElementType is int/uint and then casting to float. So you'd actually do integer rounding on the length. Something like this should fix it:
```go
return float64(sum) / float64(len(s))
```